transform/csv.go

```diff
@@ -0,0 +1,244 @@
+package transform
+
+import (
+	"archive/zip"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"unicode/utf8"
+
+	"golang.org/x/text/encoding/charmap"
+)
+
+const separator = ';'
+
+func cleanLine(l []string) ([]string, error) {
+	var c []string
+	var err error
+
+	for _, v := range l {
+		v := strings.TrimSpace(v)
+		if !utf8.ValidString(v) {
+			v, err = charmap.ISO8859_1.NewDecoder().String(v)
+			if err != nil {
+				return nil, err
+			}
+		}
+		c = append(c, strings.TrimSpace(v))
+	}
+	return c, nil
+}
+
+func (a *dataset) lineProducer(l chan<- []string, f *zip.File) error {
+	z, err := f.Open()
+	if err != nil {
+		return err
+	}
+	defer z.Close()
+
+	r := csv.NewReader(z)
+	r.Comma = separator
+	for {
+		s, err := r.Read()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return err
+		}
+		l <- s
+	}
+}
+
+func (a *dataset) lineConsumer(l chan []string) {
+	for {
+		s, ok := <-l
+		if !ok {
+			return
+		}
+
+		if err := a.csvWriter.Write(s); err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+func headersFor(a *dataset) ([]string, error) {
+	var h []string
+	switch a.kind {
+	case city, cnae, motive, nature, country, qualification:
+		h = []string{"codigo", "descricao"}
```

Comment:
Sugestão:

```suggestion
		return []string{"codigo", "descricao"}, nil
```

Nesse e nos abaixo. Vou deixar outra sugestão no final